beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -602,7 +631,7 @@ impl<T: BeaconChainTypes> DataAvailabilityCheckerInner<T> {
 
         // Check if we have all components and entire set is consistent.
         if pending_components.is_available(self.sampling_column_count, log) {
-            write_lock.put(block_root, pending_components.clone());
+            write_lock.put(block_root, pending_components.clone_without_column_recv());
```

Comment:
These clones where we delete the receiver are OK because we should be using the receiver the first time it is returned inside an available block, right?

Reply:
Yes exactly -  I added some docs to the function but I still think don't think it's immediately obvious:
https://github.com/sigp/lighthouse/blob/aabb5c101cc38618c27b62e80a74b7201b450004/beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs#L42-L44

Added some more comments above this line in https://github.com/sigp/lighthouse/pull/6756/commits/4173135299242ca420f3ba2439580d099c59f4b3:

https://github.com/sigp/lighthouse/blob/4173135299242ca420f3ba2439580d099c59f4b3/beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs#L552-L554